lib/test_formulae.rb

```diff
@@ -271,7 +271,9 @@ def bottled?(formula, no_older_versions: false)
         # up the dep tree when we encounter an `:all` bottle because
         # a formula is not bottled unless its dependencies are.
         if formula.bottle_specification.tag?(Utils::Bottles.tag(:all))
-          formula.deps.all? { |dep| bottled?(dep.to_formula, no_older_versions:) }
+          formula.deps.all? do |dep|
+            bottled?(dep.to_formula, no_older_versions: no_older_versions && (!dep.test? || dep.build?))
```

Comment:
```suggestion
            bottle_no_older_versions = no_older_versions && (!dep.test? || dep.build?)
            bottled?(dep.to_formula, no_older_versions: bottle_no_older_versions)
```
is a bit easier to follow